packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx

```diff
@@ -42,6 +43,17 @@ export const SearchIndexTemplateDropdown: React.FunctionComponent<
     [onTemplate]
   );
 
+  const templates = useMemo(() => {
+    // When vector search is supported we don't show the
+    // KNN Vector field mapping option.
+    if (isVectorSearchSupported) {
+      return ATLAS_SEARCH_TEMPLATES.filter(
+        (template) => template.name !== 'KNN Vector field mapping'
```

Comment:
nit: I know this is covered through tests, but it might be a bit nicer to handle this through a property on the `template` itself rather than to rely on the names of specific templates?

Reply:
I agree, I wasn't quite sure what I wanted to name that property though, `isReplacedByVectorSearch`?